### ISSUE TITLE
Should not reponse to right click including menubaritem/chevron in treeviewitem/...

### ISSUE THREAD

**Issue by ruihuasun** [microsoft/microsoft-ui-xaml#8041]
### Describe the bug

I don't think menubaritem/chevron in control or any other else should reponse to right mouse click.

For example, if I place some advanced operations in the menuflyout, which is called out by right mouse click. However when I click at something like chevron in treeviewitem/ menubaritem, the treenode will expand/ the menu of menubaritem will flyout. It makes no sense and will make user confused.

Also, is there any way to disable right-clicking on these controls? I try to build a inherited class based on TreeViewItem, but many methods are private which cannot be overriden.

### Steps to reproduce the bug

Right mouse click on chevron in treeviewitem / menubaritem / ....

### NuGet package version

WinUI 3 - Windows App SDK 1.2.2: 1.2.221209.1

### Windows version

Windows 10 (21H2): Build 19044
